code/src/pages/product-details/components/product-details-container/ProductDetailsContainer.tsx

```diff
@@ -0,0 +1,154 @@
+import { useIntl } from "react-intl";
+
+import AppBadge from "@/components/app-badge/AppBadge";
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { deliveryMethods as deliveryMethodsData } from "@/constants/deliveryMethods";
+import { useLocaleContext } from "@/context/i18n/I18nProvider";
+import useErrorPageRedirect from "@/hooks/use-error-page-redirect/useErrorPageRedirect";
+import { productNotFoundRedirectConfig } from "@/pages/product-details/ProductsDetailsPage.constants";
+import { useGetUserProductByIdQuery } from "@/store/api/productsApi";
+import formatPrice from "@/utils/format-price/formatPrice";
+import getCategoryFromTags from "@/utils/get-category-from-tags/getCategoryFromTags";
+import isErrorWithStatus from "@/utils/is-error-with-status/isErrorWithStatus";
+
+import "@/pages/product-details/components/product-details-container/ProductDetailsContainer.scss";
+
+type ProductDetailsContainerProps = {
+  productId: string;
+};
+
+const ProductDetailsContainer = ({
+  productId
+}: ProductDetailsContainerProps) => {
+  const { renderRedirectComponent } = useErrorPageRedirect();
+  const { locale } = useLocaleContext();
+  const { formatMessage } = useIntl();
+  const {
+    data: product,
+    isLoading,
+    error
+  } = useGetUserProductByIdQuery({
+    productId,
+    lang: locale
+  });
+
+  // @TODO: replace with actual loading fallback
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  const isNotFoundOnServer = isErrorWithStatus(error) && error.status === 404;
+
+  if (isNotFoundOnServer) {
+    return renderRedirectComponent(productNotFoundRedirectConfig);
+  }
+
+  // @TODO: add error handling
+  if (!product || error) {
+    return <div>Error...</div>;
```

Comment:
maybe for now good to hadnle some `<AppTypography translationKey="error.label" />` for error. 
like our previously examples.
Also nice to render some `AppLoader` for loading instead text above I mean to handle `isLoading`

Reply:
As for error, @D1nviz assigned to an issue to create reusable error layout that will look nicer. I think we can also create issue for fixing all loader on all dashboard tabs, because for now on orders page for example we have only `<div>Loading...</div>` for now.

I attached a screenshot below:

![image](https://github.com/user-attachments/assets/b42ad5d9-2c43-4133-8f5d-7755cb5a0033)